### DISCUSSION TITLE
how to generate a "real" dynamic url ? not in "pre" !!!

### DISCUSSION THREAD

**Discussion by LLLPY** [vuejs/vitepress#3348] [Q&A] [answered]
recently, i'm going to build a blog system in vitepress.exactly i‘m upgrade my blog system with it. but i have a few problems when i work on it.

1.i can't get the new blog i wrote until i restart my server.
i get my dynamic routes from remote data, each id corresponds to a blog post.it's good when i do nothing.
<img width="1512" alt="image" src="https://github.com/vuejs/vitepress/assets/61370857/726f3def-d68d-4ff9-92cf-05496afe37ec">
but when i add a new blog on my remote server, and i wanna visit it on vitepress, it turns out:404, i know the reason, because i add it on my remote server,but i didn't load in my front system(vitepress). So it there a way to solve it?
<img width="1512" alt="image" src="https://github.com/vuejs/vitepress/assets/61370857/9adee7a1-e6c6-4249-b9a3-8fd31a66ad68">

2.how to generate a dynamic nav menu.
i only want to show the "写作" and "个人中心" button when user was logged in and "退出登录" when user was logged out.
<img width="1512" alt="image" src="https://github.com/vuejs/vitepress/assets/61370857/1cb38d96-0576-4f4e-a9da-c7afeb7aa78b">

in addition my blog system site is: http://www.lll.plus . i'm building it. everyone is welcome to visit it.and if you have are interest in it, please contact me.maybe we can build it together hhhhh.

**Comment by maicss** [comment, marked as answer]
This is a **static** content generator. For dynamic content I suggest you check Nuxtjs

这是个静态内容生成器，如果🈶动态内容的需要，可以使用 nuxtjs

**Reply by LLLPY** [reply]
how to use it in vitepress?

**Reply by maicss** [reply]
You can't

**Comment by LLLPY** [comment]
is there any to speed up at the first time?
